nixos/modules/tasks/flake-auto-upgrade.nix

```diff
@@ -0,0 +1,306 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  nixCommand =
+    "${pkgs.nix}/bin/nix --extra-experimental-features 'nix-command flakes'";
+  mkBuilds = attrs: concatMapStrings mkBuild attrs;
+  mkBuild = attr:
+    if builtins.typeOf attr == "string" then ''
+      if ! ${nixCommand} build .#${attr};
+      then
+        echo ${attr} >> fail.log
+      fi
+    '' else ''
+      if ! ${nixCommand} build .#${attr.attr};
+      then
+        echo ${attr.attr} >> fail.log
+        ${mkBuilds attr.onFail}
+      fi
+    '';
+  mkServices = name: cfg:
+    nameValuePair name {
+      description = "Nix updater";
+      environment = mkMerge [
+        (mkIf (cfg.ssh != null) {
+          GIT_SSH_COMMAND = let
+            knownHostsFile = pkgs.writeText "knownHostsFile" cfg.ssh.hostKey;
+          in "${pkgs.openssh}/bin/ssh -i $CREDENTIALS_DIRECTORY/sshKey -o GlobalKnownHostsFile=${knownHostsFile}";
+        })
+        (mkIf (cfg.credentials != null) {
+          GIT_ASKPASS = pkgs.writeShellScript "askpass" ''
+            case "$1" in
+                Username*)
+                  echo ${cfg.credentials.user}
+                  ;;
+                Password*)
+                  cat $CREDENTIALS_DIRECTORY/passwordFile
+                  ;;
+            esac
+          '';
+        })
+        {
+          EMAIL = "<>";
+          HOME = "/var/lib/${name}";
+        }
+      ];
+      path =
+        mkMerge [ (mkIf cfg.gitCrypt.enable [ pkgs.git-crypt ]) [ pkgs.git ] ];
+      serviceConfig = mkMerge [
+        {
+          type = "oneshot";
```

Comment:
`type` should be outside of `serviceConfig`

Reply:
That doesn't exist. Other services are using uppercase `Type`